src/view/domconverter.js

```diff
@@ -651,4 +654,29 @@ export default class DomConverter {
 	isDocumentFragment( node ) {
 		return node && node.nodeType == Node.DOCUMENT_FRAGMENT_NODE;
 	}
+
+	/**
+	 * Returns `true` if given selection is a backward selection, that is, if it's `focus` is before `anchor`.
+	 *
+	 * @param {Selection} DOM Selection instance to check.
+	 * @return {Boolean}
+	 */
+	isDomSelectionBackward( selection ) {
```

Comment:
It's definitely premature, but I know that browsers hate ranges because they are "live". So I'm really worried that creating a range instance must create many listeners and will be slow, which in case of selection rendering may have a negative impact on the performance, but very hard to estimate. Another thing which is hard to estimate is memory pressure caused by ranges. 

But when I started writing down how an algorithm using `compareDocPos` would look like I understood why you used ranges :D. Let's not optimise it now.